src/expr/src/expr/expr_some_all.rs

```diff
@@ -0,0 +1,195 @@
+        let bitmap = data_chunk.get_visibility_ref();
+        let mut num_array = Vec::with_capacity(data_chunk.capacity());
+
+        let arr_right_inner = arr_right.as_list();
+        let ArrayMeta::List { datatype } = arr_right_inner.array_meta() else {
+            unreachable!()
+        };
+        let capacity = arr_right_inner
+            .iter()
+            .flatten()
+            .map(|list_ref| list_ref.flatten().len())
+            .sum();
+
+        let mut unfolded_arr_left_builder = arr_left.create_builder(capacity);
+        let mut unfolded_arr_right_builder = datatype.create_array_builder(capacity);
+
+        let mut unfolded_left_right =
+            |left: Option<ScalarRefImpl<'_>>,
+             right: Option<ScalarRefImpl<'_>>,
+             num_array: &mut Vec<usize>| {
+                let datum_left = left.map(|s| s.into_scalar_impl());
+                let datum_right = right.map(|s| s.into_scalar_impl());
+                if datum_right.is_none() {
+                    num_array.push(0);
+                    return;
+                }
+
+                let datum_right = datum_right.unwrap();
+                match datum_right {
+                    ScalarImpl::List(array) => {
+                        let len = array.values().len();
+                        num_array.push(len);
+                        unfolded_arr_left_builder.append_datum_n(len, datum_left.as_ref());
+                        for item in array.values() {
+                            unfolded_arr_right_builder.append_datum(item.as_ref());
+                        }
+                    }
+                    _ => unreachable!(),
+                }
+            };
+
+        match bitmap {
+            Some(bitmap) => {
+                for ((left, right), visible) in
+                    multizip((arr_left.iter(), arr_right.iter())).zip_eq(bitmap.iter())
+                {
+                    if !visible {
+                        num_array.push(0);
+                        continue;
+                    }
+                    unfolded_left_right(left, right, &mut num_array);
+                }
+            }
+            None => {
+                for (left, right) in multizip((arr_left.iter(), arr_right.iter())) {
+                    unfolded_left_right(left, right, &mut num_array);
+                }
+            }
+        }
+
+        let data_chunk = DataChunk::new(
+            vec![
+                unfolded_arr_left_builder.finish().into(),
+                unfolded_arr_right_builder.finish().into(),
+            ],
+            capacity,
+        );
+
+        let func_results = self.func.eval(&data_chunk)?;
+        let mut func_results_iter = func_results.as_bool().iter();
+        Ok(Arc::new(
+            num_array
+                .into_iter()
+                .map(|num| {
+                    self.resolve_boolean_vec(func_results_iter.by_ref().take(num).collect_vec())
```

Comment:
When `num` is 0, there are 3 reasons:
* The corresponding row is invisible.
* The rhs array is `null`.
* The rhs array is empty (`array[]`).

Note that PostgreSQL returns different results for a null array and empty array:
```
test=# select 1 < ALL(array[]::int[]);
 ?column? 
----------
 t
(1 row)

test=# select 1 < SOME(array[]::int[]);
 ?column? 
----------
 f
(1 row)

test=# select 1 < ALL(null);
 ?column? 
----------
 
(1 row)

test=# select 1 < SOME(null);
 ?column? 
----------
 
(1 row)
```